README.md

```diff
@@ -1,28 +1,204 @@
 # url-mapper
-Take a URL and map to functions, parsing params
+Two way `URL` <==> `route(params)` converter with mapper.
 
-```js
-import route from 'url-mapper';
+[![NPM version][npm-image]][npm-url]
+[![Build status][travis-image]][travis-url]
+[![Test coverage][coveralls-image]][coveralls-url]
+
+## Instalation
+
+`npm install url-mapper --save`
+
+## Usage
+
+### Overview
+
+The main purpose of `url-mapper` is match given `URL` to one of the `routes`.
+It will return matched route (key and associated value) and parsed parameters.
+You can associate with route anything you want: function, React component or just plain object.
+
+`url-mapper` is helpful when creating router packages for frameworks or can be used as router itself.
+It allows you to outsource work with url (mapping, parsing, stringifying) and concentrate on wiring up things related to your favorite framework.
+
+### Example
 
-const someFunc = function (data) {
-  data // {url: '/foo/123?foo=bar', path: '/foo', params: {id: '123'}, query: {foo: 'bar'}}
-};
+```js
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Mapper from 'url-mapper';
+import { CoreApp, ComponentA, ComponentB, Component404 } from './components';
 
-const notFoundFunc = function (data) { };
+const urlMapper = Mapper();
 
-route(location.origin + '/foo/123', {
-  '/foo/:id': someFunc,
-  '*': notFoundFunc
+var matchedRoute = urlMapper('/bar/baz/:42', { // routable part of url
+  '/foo/:id': ComponentA,
+  '/bar/:list/:itemId': ComponentB,
+  '*': Component404
 });
+
+if (matchedRoute) {
+  const Component = matchedRoute.match; // ComponentB
+  const props = matchedRoute.values; // { list: 'baz', itemId: 42 }
+
+  ReactDOM.render(
+    <CoreApp>
+      <Component {...props} />
+    </CoreApp>
+  );
+}
 ```
-**url-mapper** passes an object representing the parsed URL to your route callbacks. Given the URL `http://www.bigapp.com/foo/123?bar=baz` and a matching route `/foo/:id`, your callbacks will receive:
+
+See [`cerebral-router`](https://github.com/christianalfoni/cerebral-router) as example of building own router solution on top of `url-mapper`.
+
+## API
+
+### Main module
+
+At top level `url-mapper` module exports factory which returns default implementation of `URL` <==> `route(params)` converter.
```

Comment:
At the top level `url-mapper` module exports a factory which returns the default implementation of a `URL` <==> `route(params)` converter.